include/af/blas.h

```diff
@@ -7,165 +7,153 @@
  * http://arrayfire.com/licenses/BSD-3-Clause
  ********************************************************/
 
-/** \file blas.h
- *
- * Contains BLAS related functions
- *
- * Contains functions for basic BLAS functionallity
- */
-
 #pragma once
-
 #include <af/defines.h>
 
 #ifdef __cplusplus
 namespace af
 {
     class array;
     /**
-        \brief Matrix multiply of two arrays
-
-        \copydetails blas_func_matmul
-
-        \param[in] lhs The array object on the left hand side
-        \param[in] rhs The array object on the right hand side
-        \param[in] optLhs Transpose left hand side before the function is performed
-        \param[in] optRhs Transpose right hand side before the function is performed
-        \return The result of the matrix multiplication of lhs, rhs
+       C++ Interface to multiply two matrices.
 
-        \note optLhs and optRhs can only be one of \ref AF_MAT_NONE, \ref
-              AF_MAT_TRANS, \ref AF_MAT_CTRANS \note This function is not supported
-              in GFOR
-
-        \note <b> The following applies for Sparse-Dense matrix multiplication.</b>
-        \note This function can be used with one sparse input. The sparse input
-              must always be the \p lhs and the dense matrix must be \p rhs.
-        \note The sparse array can only be of \ref AF_STORAGE_CSR format.
-        \note The returned array is always dense.
-        \note \p optLhs an only be one of \ref AF_MAT_NONE, \ref AF_MAT_TRANS,
-              \ref AF_MAT_CTRANS.
-        \note \p optRhs can only be \ref AF_MAT_NONE.
+       \copydetails blas_func_matmul
 
-        \ingroup blas_func_matmul
+       `optLhs` and `optRhs` can only be one of \ref AF_MAT_NONE,
+       \ref AF_MAT_TRANS, \ref AF_MAT_CTRANS.
+       
+       This function is not supported in GFOR.
+
+       \note <b>The following applies for Sparse-Dense matrix multiplication.</b>
+       \note This function can be used with one sparse input. The sparse input
+             must always be the \p lhs and the dense matrix must be \p rhs.
+       \note The sparse array can only be of \ref AF_STORAGE_CSR format.
+       \note The returned array is always dense.
+       \note \p optLhs an only be one of \ref AF_MAT_NONE, \ref AF_MAT_TRANS,
+             \ref AF_MAT_CTRANS.
+       \note \p optRhs can only be \ref AF_MAT_NONE.
+
+       \param[in] lhs    input array on the left-hand side
+       \param[in] rhs    input array on the right-hand side
+       \param[in] optLhs transpose the left-hand side prior to multiplication
+       \param[in] optRhs transpose the right-hand side prior to multiplication
+       \return    `lhs` * `rhs`
 
-     */
+       \ingroup blas_func_matmul
+    */
     AFAPI array matmul(const array &lhs, const array &rhs,
                        const matProp optLhs = AF_MAT_NONE,
                        const matProp optRhs = AF_MAT_NONE);
 
     /**
-       \brief Matrix multiply of two arrays
+       C++ Interface to multiply two matrices.
```

Comment:
This function should discuss how the second parameter is transposed.

Reply:
Not sure what you mean. The optLhs and optRhs parameters are described.